File: Frontend/VIAProMa/Assets/Scripts/DataModel/RequirementsBazaar/ReqBazUser.cs
using i5.VIAProMa.UI.ListView.Core;
using System;

namespace i5.VIAProMa.DataModel.ReqBaz
{
    /// <summary>
    /// Represents a user in the Requirements Bazaar
    /// </summary>
    [Serializable]
    public class ReqBazUser : IListViewItemData, IUninitializable
    {

        /// <summary>
        /// The values of the requirements bazaar user
        /// </summary>
        public int id;
        public string userName;
        public string firstName;
        public string lastName;
        public bool admin;
        public long las2peerId;
        public string profileImage;
        public bool emailLeadSubscription;
        public bool emailFollowSubscription;

        /// <summary>
        /// Whether or not the user has been initialized
        /// </summary>
        public bool IsUninitialized
        {
            get
            {
                if (id == 0 && las2peerId == 0)
                {
                    return true;
                }
                else
                {
                    return false;
                }
            }
        }
    }
}

File: Backend/immersive_project_management_service/src/main/java/i5/las2peer/services/immersiveProjectManagementService/i5/las2peer/services/immersiveProjectManagementService/dataModel/requirementsBazaar/ReqBazUser.java
package i5.las2peer.services.immersiveProjectManagementService.i5.las2peer.services.immersiveProjectManagementService.dataModel.requirementsBazaar;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

/**
 * Created by bened on 09.05.2019.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class ReqBazUser {
    public int id;
    public String userName;
    public String firstName;
    public String lastName;
    public boolean admin;
    public long las2peerId;
    public String profileImage;
    public boolean emailLeadSubscription;
    public boolean emailFollowSubscription;

    @Override
    public boolean equals(Object obj) {
        if (!(obj instanceof ReqBazUser))
        {
            return false;
        }
        return this.id == ((ReqBazUser) obj).id;
    }
}
